include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp

```diff
@@ -381,6 +381,14 @@ namespace SuperiorMySqlpp
                     auto&& binding = *bindingsIt;
                     auto&& resultMetadata = *metadataIt;
 
+                    // It's needed to check buffer and length pointers to detect uninitialized binds.
+                    // Buffer may point to nullptr in case bind is initialized as ArrayBase of zero length
+                    // but the `length` will point to variable with buffer length in such case.
+                    if (bindingsIt->buffer == nullptr && bindingsIt->length == nullptr)
```

Comment:
Actually, we may need to ignore one or more result column - so the check might be probably for binding type, but then, this would need to have special ignored result type and also we must check, if underlaying mysql c-lib support ignoring result. This would require more logic, so i will create new issue for this.
Anyway shouldn't be enough to check only buffer length only?

Reply:
Also, it's kind of strange to check binding engagement by this kind of hack :)

Reply:
It's needed to check both conditions. The `length` can be `nullptr` in case the numeric type is binded.

See https://dev.mysql.com/doc/refman/5.7/en/c-api-prepared-statement-data-structures.html
> `length` is ignored for numeric and temporal data types because the `buffer_type` value determines the length of the data value.